plugin/trino-teradata/src/test/java/io/trino/plugin/teradata/integration/clearscape/TeradataHttpClient.java

```diff
@@ -0,0 +1,166 @@
+                .GET()
+                .build();
+        var httpResponse =
+                handleCheckedException(() -> httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString()));
+        return handleHttpResponse(httpResponse, new TypeReference<>() {});
+    }
+
+    public CompletableFuture<Void> deleteEnvironment(DeleteEnvironmentRequest deleteEnvironmentRequest, String token)
+    {
+        var httpRequest = HttpRequest.newBuilder(URI.create(baseUrl + "/environments/" + deleteEnvironmentRequest.name()))
+                .headers(AUTHORIZATION, BEARER + token)
+                .DELETE()
+                .build();
+
+        // start async and ignore returned future
+        httpClient.sendAsync(httpRequest, HttpResponse.BodyHandlers.ofString());
+        return CompletableFuture.completedFuture(null);
+    }
+
+    public void startEnvironment(EnvironmentRequest environmentRequest, String token)
+    {
+        var requestBody = handleCheckedException(() -> objectMapper.writeValueAsString(environmentRequest.request()));
+        getVoidCompletableFuture(environmentRequest.name(), token, requestBody);
+    }
+
+    public void stopEnvironment(EnvironmentRequest environmentRequest, String token)
+    {
+        var requestBody = handleCheckedException(() -> objectMapper.writeValueAsString(environmentRequest.request()));
+        getVoidCompletableFuture(environmentRequest.name(), token, requestBody);
+    }
+
+    private void getVoidCompletableFuture(String name, String token, String jsonPayLoadString)
+    {
+        HttpRequest.BodyPublisher publisher = HttpRequest.BodyPublishers.ofString(jsonPayLoadString);
+        var httpRequest = HttpRequest.newBuilder(URI.create(baseUrl + "/environments/" + name))
+                .headers(AUTHORIZATION, BEARER + token, CONTENT_TYPE, APPLICATION_JSON)
+                .method("PATCH", publisher)
+                .build();
+
+        httpClient.sendAsync(httpRequest, HttpResponse.BodyHandlers.ofString());
+    }
+
+    private <T> T handleHttpResponse(HttpResponse<String> httpResponse, TypeReference<T> typeReference)
+    {
+        var body = httpResponse.body();
+        if (httpResponse.statusCode() >= 200 && httpResponse.statusCode() <= 299) {
+            return handleCheckedException(() -> {
+                if (typeReference.getType().getTypeName().equals(Void.class.getTypeName())) {
+                    return null;
+                }
+                else {
+                    return objectMapper.readValue(body, typeReference);
+                }
+            });
+        }
+        else if (httpResponse.statusCode() >= 400 && httpResponse.statusCode() <= 499) {
+            throw new Error4xxException(httpResponse.statusCode(), body);
+        }
+        else if (httpResponse.statusCode() >= 500 && httpResponse.statusCode() <= 599) {
+            throw new Error5xxException(httpResponse.statusCode(), body);
+        }
+        else {
```

Comment:
Redundant else.

Reply:
Merged Error4xxException and Error5xxException to single exception class `ClearScapeServiceException`.